packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -0,0 +1,64 @@
+import cloudformation = require('@aws-cdk/aws-cloudformation');
+import lambda = require('@aws-cdk/aws-lambda');
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import { ISource } from './source';
+
+const handlerCodeBundle = require.resolve('@aws-cdk/aws-s3-deployment-handler/dist/lambda.zip');
+
+export interface BucketDeploymentProps {
+  /**
+   * The source from which to deploy the contents of this bucket.
+   */
+  source: ISource;
+
+  /**
+   * The S3 bucket to sync the contents of the zip file to.
+   */
+  destinationBucket: s3.BucketRef;
+
+  /**
+   * Key prefix in desination.
+   * @default No prefix (source == dest)
+   */
+  destinationKeyPrefix?: string;
+
+  /**
+   * If this is enabled, files in destination bucket/prefix will not be deleted
+   * when the resource is deleted or removed from the stack.
+   *
+   * @default false (when resource is deleted, files are deleted)
+   */
+  retainOnDelete?: boolean;
+}
+
+export class BucketDeployment extends cdk.Construct {
+  constructor(parent: cdk.Construct, id: string, props: BucketDeploymentProps) {
+    super(parent, id);
+
+    const handler = new lambda.SingletonFunction(this, 'CustomResourceHandler', {
+      uuid: '8693BB64-9689-44B6-9AAF-B0CC9EB8756C',
```

Comment:
Also provide a `purpose`, it'll make the Lambda logs more findable in the CloudWatch console.

Reply:
Done. thanks